ignite/metrics/metric.py

```diff
@@ -108,14 +118,37 @@ def _sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union
         else:
             raise TypeError("Unhandled input type {}".format(type(tensor)))
 
-        # synchronize and reduce
-        dist.barrier()
-        dist.all_reduce(tensor)
+        reduction(tensor)
 
         if tensor_to_number:
             return tensor.item()
         return tensor
 
+    @staticmethod
+    def _tpu_reduce(tensor: Union[torch.Tensor, numbers.Number]) -> None:
+        xm.all_reduce("sum", [tensor,], torch.float)
```

Comment:
I'm not sure about `torch.float` here. The API is a bit different
http://pytorch.org/xla/release/1.5/index.html#torch_xla.core.xla_model.all_reduce

Reply:
oops, this was a mistake when cleaning up the abstraction. that's supposed to happen in the number/tensor conversion.